launcher/updatedialog_moc.cpp

```diff
@@ -0,0 +1,143 @@
+/*
+ * updatedialog_moc.cpp, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+#include "StdInc.h"
+#include "updatedialog_moc.h"
+#include "ui_updatedialog_moc.h"
+
+#include "../lib/CConfigHandler.h"
+#include "../lib/GameConstants.h"
+
+#include <QNetworkAccessManager>
+#include <QNetworkReply>
+#include <QNetworkRequest>
+
+UpdateDialog::UpdateDialog(QWidget *parent, bool calledManually) :
+	QDialog(parent),
+	ui(new Ui::UpdateDialog),
+	calledManually(calledManually)
+{
+	ui->setupUi(this);
+	
+	if(calledManually)
+	{
+		setWindowModality(Qt::ApplicationModal);
+		show();
+	}
+	
+	connect(ui->closeButton, SIGNAL(clicked()), this, SLOT(close()));
+	
+	if(settings["launcher"]["updateOnStartup"].Bool() == true)
+		ui->checkOnStartup->setCheckState(Qt::CheckState::Checked);
+	
+	currentVersion = GameConstants::VCMI_VERSION;
+	
+	setWindowTitle(QString::fromStdString(currentVersion));
+	
+#ifdef VCMI_WINDOWS
+	platformParameter = "windows";
+#elif defined(VCMI_MAC)
+	platformParameter = "macos";
+#elif defined(VCMI_IOS)
+	platformParameter = "ios";
+#elif defined(VCMI_ANDROID)
+	platformParameter = "android";
+#elif defined(VCMI_UNIX)
+	platformParameter = "linux";
+#endif
+	
+	QString url = QString::fromStdString(settings["launcher"]["updateConfigUrl"].String());
+	
+	QScopedPointer<QNetworkAccessManager> manager(new QNetworkAccessManager);
+	
+	QNetworkReply *response = manager->get(QNetworkRequest(QUrl(url)));
+	
+	QObject::connect(response, &QNetworkReply::finished, [&, response]{
+		response->deleteLater();
+		response->manager()->deleteLater();
+		if(response->error() != QNetworkReply::NoError)
+		{
+			ui->versionLabel->setStyleSheet("QLabel { background-color : red; color : black; }");
+			ui->versionLabel->setText("Network error");
+			ui->plainTextEdit->setPlainText(response->errorString());
+			return;
+		}
+		
+		auto byteArray = response->readAll();
+		JsonNode node(byteArray.constData(), byteArray.size());
```

Comment:
as the final goal is to throw JSON data into Qt objects, it might be better to use `QJsonDocument` to deal with JSON

Reply:
We use JsonNode in other places of vcmilauncher so It make sense to keep one mechanism rather t

Reply:
It's part of the future work